pages/en/lb4/Mixin.md

```diff
@@ -0,0 +1,119 @@
+---
+lang: en
+title: 'Mixin'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Mixin.html
+summary:
+---
+
+It is a commonly used JavaScript/TypeScript strategy to extend a class with new properties and methods.
+
+*question: does our mixinBuilder allow interface extends classes?*
+
+## Mixin Builder
+
+LoopBack provides a mixin builder function `MixinBuilder` to easily extend a class 
+with one or multiple mixins,
+you can call it to define the new extended class with fluent syntaxed API like:
+
+```ts
+import {MixiBuilder} from '@loopback/repository';
+import {BaseClass} from '<path_exports_BaseClass>';
+import {FooMixin, BarMixin} from '<path_exports_Mixins>';
+
+let newClass = MixinBuilder.mix(baseClass).with(FooMixin, BarMixin);
+```
+
+## Define Mixin
```

Comment:
I wonder if we should walk readers through this section first before `Mixin Builder` section

Reply:
I also think of the sequence when wrote it. Eventually decided to introduce `MixinBuilder` first:

`MixinBuilder.mix(baseClass).with(FooMixin, BarMixin)` is a very straight forward syntax to give a new  mixin learner a big picture of what is the relation between class and mixin, then they go into section#define-mixin to dig more into implementation and usage. And to tell people already know mixin how `MixinBuilder` provides a prettier syntax to apply mixins.